hw/ip/spi_host/doc/_index.md

```diff
@@ -1288,7 +1288,168 @@ Highlights for this unit:
+The SPI_HOST IP is however suitable for interacting with any number of SPI devices, and the same mode of operation can be used for any SPI device.
+
+## Initializing the IP
+
+### Per-target Configuration
+
+The {{< regref "CONFIGOPTS" >}} multi-register must be programmed to reflect the requirements of the attached target devices.
+As such these registers can be programmed once at initialization, or whenever a new device is connected (e.g., via changes in the external pin connections, or changes in the pinmux configuration).
+The proper settings for the {{< regref "CONFIGOPTS" >}} fields (e.g., CPOL and CPHA, clock divider, ratios, and other timing or sampling requirements) will all depend on the specific device attached as well as the board level delays.
+
+### Interrupt configuration
+
+The next step is to configuration the interrupts for the SPI_HOST.
+This should also be done at initialization using the following register fields:
+- The {{< regref "ERROR_ENABLE" >}} register should be configured to indicate what types of error conditions (if any) should be ignored to not trigger an interrupt.
+At reset, these fields are all set indicating that all error classes trigger an interrupt.
+
+- For interrupt driven I/O the {{< regref "EVENT_ENABLE" >}} register must be configured to select the desired event interrupts to signal the desired conditions (e.g. "FIFO empty", "FIFO at the watermark level", or "ready for next command segment").
+By default, this register is all zeros, meaning all event interrupts are disabled, and thus all transactions must be managed by polling the status register.
+   - When using the FIFO watermarks to send interrupts, the watermark levels must be set via the {{< regref "CONTROL.RX_WATERMARK" >}} and {{< regref "CONTROL.TX_WATERMARK" >}} fields.
+
+- The event and error interrupts must finally be enabled using the {{< regref "INTR_ENABLE" >}} register.
+
+### Enabling the SPI_HOST
+
+The IP must be enabled before sending the first command by asserting the {{< regref "CONTROL.SPIEN" >}} bit.
+
+## Issuing Transactions
+
+As mentioned above, each command is typically specified in three phases: loading the TX data, specifying the command segments/format, and reading the RX data.
+In principle, the first two steps can be performed in either order.
+If the SPI_HOST does not find any data to transmit it will simply stall until data is inserted.
+Meanwhile, the RX data is only available after the command format has been specified and processed by the FSM.
+
+For longer transactions, with data larger than the capacity of the FIFOs, the command sequence may become more complex.
+For instance, to send 1024 bytes of data in a single command, the TX data may need to be loaded several times if using a 256-byte FIFO.
+So in this example the programming sequence will consist of at least four iterations of entering TX data and waiting for the TX FIFO to drain.
```

Comment:
Thinking about how to make this section flow a bit better.

I think the first sentence is fine: the use of "transaction" describes what you want to happen and the use of "command sequence" is describing what the programmer will do to the IP block.

In the second sentence, I think "command" should be replaced with "transaction".

I might start the third sentence with "In this instance, ..." because the prior sentence starts with "For instance, ...".

Reply:
```suggestion
For longer transactions, with data larger than the capacity of the FIFOs, the command sequence may become more complex.
For instance, to send 1024 bytes of data in a single transaction, the TX data may need to be loaded several times if using a 256-byte FIFO.
In this instance, the programming sequence will consist of at least four iterations of entering TX data and waiting for the TX FIFO to drain.
```